### PR TITLE
Ability to pass extra labels to test jobs if defined in variable file

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Adds the ability to pass parameter LABEL_ADDITION to
  the test pipelines when extra_test_labels is defined
  in the variable file.
- Refactor some of the code that initializes the test
  targets.
- Remove default test targets from variable file. Simplifies
  processing of tests_targets. All builds are setup to specify
  the test targets to run so this is unnecessary code.
- Keep sanity and extended definition mapping to functional
  testing respectively.
    
[skip ci]
Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Depends #8387 
@vsebe for review
Jenkins test sanity plinux jdk8

**Comment by AdamBrousseau** [comment]
@pshipton note the sanity suite was excluded from the cmake build.

**Comment by AdamBrousseau** [comment]
I have added #8387 to this PR and will close the other one.

File: buildenv/jenkins/common/variables-functions.groovy [hunk 1]
```diff
@@ -637,14 +637,38 @@ def get_date() {
...
-    if (!TESTS_TARGETS) {
-        // set default TESTS_TARGETS for pipeline job (fetch from variables file)
-        TESTS_TARGETS = get_default_test_targets()
+    TESTS_TARGETS = (params.TESTS_TARGETS) ? params.TESTS_TARGETS.trim() : "none"
+    TARGET_NAMES = []
+
+    if (TESTS_TARGETS != 'none') {
+        for (target in TESTS_TARGETS.replaceAll("\\s","").toLowerCase().tokenize(',')) {
+            switch (target) {
+                case "sanity":
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions.groovy]: Please combine the 2 case statements, e.g. `case ['sanity', 'extended']`.

File: buildenv/jenkins/common/variables-functions.groovy [hunk 2]
```diff
@@ -637,14 +637,38 @@ def get_date() {
...
-* Set TESTS_TARGETS, indicating the level of testing
-* and sets TEST_FLAG for all targets if defined in variable file
+* Set TESTS_TARGETS, indicating the level of testing.
 */
 def set_test_targets() {
-    TESTS_TARGETS = params.TESTS_TARGETS
-    if (!TESTS_TARGETS) {
-        // set default TESTS_TARGETS for pipeline job (fetch from variables file)
-        TESTS_TARGETS = get_default_test_targets()
+    TESTS_TARGETS = (params.TESTS_TARGETS) ? params.TESTS_TARGETS.trim() : "none"
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions.groovy]: "none" and 'none' be consistent. As there is no no interpolation single quotes will do.

File: buildenv/jenkins/common/variables-functions.groovy [hunk 3]
```diff
@@ -637,14 +637,38 @@ def get_date() {
...
+    }
+}
+
+/*
+* Set TEST_FLAG for all targets if defined in variable file.
+* Set EXCLUDED_TESTS if defined in variable file.
+* Set EXTRA_TEST_LABELS map if defined in variable file.
+*/
+def set_test_misc() {
+    EXTRA_TEST_LABELS = [:]
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions.groovy]: Extra test labels can be added only in the variables file, no build parameters available. Should users be able to set them in the UI when a build is launched?

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy [hunk 4]
```diff
@@ -136,7 +136,7 @@ SHORT_NAMES = ['all' : ['ppc64le_linux','ppc64le_linux_xl','s390x_linux','s390x_
 echo "Initialize all PARAMETERS..."
 SETUP_LABEL = (params.SETUP_LABEL) ? params.SETUP_LABEL : "worker"
 echo "Setup SETUP_LABEL:'${SETUP_LABEL}'"
-TESTS_TARGETS = (params.TESTS_TARGETS) ? params.TESTS_TARGETS : ""
+TESTS_TARGETS = (params.TESTS_TARGETS) ? params.TESTS_TARGETS : "none"
```
Comment (vsebe) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: `TESTS_TARGETS` are set by `set_test_targets()` function in variables-functions.groovy. This  is not necessary.

File: buildenv/jenkins/common/variables-functions.groovy (hunk 1 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy]: done

File: buildenv/jenkins/common/variables-functions.groovy (hunk 2 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy]: done

File: buildenv/jenkins/common/variables-functions.groovy (hunk 3 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy]: I don't think it's a necessary use case. Can be added later if needed. Use in a grinder would be more likely.

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy (hunk 4 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy]: Done

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional plinux jdk11

**Comment by AdamBrousseau** [comment]
There's a bug setting the targets in PRs. Let me fix.

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional plinux,xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
Ok I think it's fixed. Not sure if you want to take another look @vsebe .

**Comment by pshipton** [comment]
jenkins test sanity.functional plinux,xlinuxcm jdk11

**Comment by pshipton** [comment]
The cmake build is showing
```
17:16:01  The 'sanity.functional' test suite will be excluded
[Pipeline] End of Pipeline
java.lang.IllegalStateException: unexpected continue statement
	at com.cloudbees.groovy.cps.impl.CallEnv.getContinueAddress(CallEnv.java:106)
	at com.cloudbees.groovy.cps.impl.ProxyEnv.getContinueAddress(ProxyEnv.java:56)
	at com.cloudbees.groovy.cps.impl.ProxyEnv.getContinueAddress(ProxyEnv.java:56)
	at com.cloudbees.groovy.cps.impl.ProxyEnv.getContinueAddress(ProxyEnv.java:56)
	at com.cloudbees.groovy.cps.impl.ContinueBlock.eval(ContinueBlock.java:21)
	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
	at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
	at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:186)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:370)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$200(CpsThreadGroup.java:93)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:282)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:270)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:66)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Finished: FAILURE
```

**Comment by AdamBrousseau** [comment]
I'm guessing it's because I changed from a for loop to an each loop. I will fix tonight and relaunch. Good catch.

**Comment by AdamBrousseau** [comment]
previous build
- https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2564/
   - https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK11_ppc64le_linux/1546
      - https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_ppc64le_linux_Personal/459/

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional xlinuxcm jdk11

**Comment by AdamBrousseau** [comment]
lgtm now

**Comment by pshipton** [comment]
Something's not working right. This cmake sanity test from a PR build wasn't expected.
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_x86-64_linux_cm_Personal/3/

**Comment by AdamBrousseau** [comment]
I think the excludes aren't being matched properly because the string types aren't the same
```
00:31:14.811  sanity.functional
[Pipeline] echo
00:31:14.813  class java.lang.String
[Pipeline] echo
00:31:14.816  sanity.functional
[Pipeline] echo
00:31:14.818  class org.codehaus.groovy.runtime.GStringImpl
```
String in the exclude is a `String`, target is a `GStringImpl` which are note equal.
Working on a fix.

**Pull request merged**